pkg/opentelemetry-mapping-go/otlp/metrics/metrics_translator.go

```diff
@@ -141,15 +139,99 @@ func NewTranslator(set component.TelemetrySettings, attributesTranslator *attrib
 	}
 
 	cache := newTTLCache(cfg.sweepInterval, cfg.deltaTTL)
+	logger := set.Logger.With(zap.String("component", "metrics translator"))
 
-	return &Translator{
+	t := &defaultTranslator{
 		prevPts:              cache,
-		logger:               set.Logger.With(zap.String("component", "metrics translator")),
+		logger:               logger,
 		attributesTranslator: attributesTranslator,
 		cfg:                  cfg,
+	}
+	// Use custom mapper if provided, otherwise create a DefaultMapper
+	if cfg.customMapper != nil {
+		t.mapper = cfg.customMapper
+	} else {
+		t.mapper = newDefaultMapper(cache, logger, cfg)
+	}
+	return t, nil
+}
+
+// NewTranslator creates a new translator with given options.
+//
+// Deprecated: Use [NewDefaultTranslator] instead.
+func NewTranslator(set component.TelemetrySettings, attributesTranslator *attributes.Translator, options ...TranslatorOption) (*Translator, error) {
+	d, err := NewDefaultTranslator(set, attributesTranslator, options...)
+	if err != nil {
+		return nil, err
+	}
+	return &Translator{
+		Provider: d,
+		logger:   set.Logger,
 	}, nil
 }
 
+// getMapper returns the underlying Mapper implementation.
+// This is useful for testing or for direct access to mapping methods.
+func (t *defaultTranslator) getMapper() Mapper {
+	return t.mapper
+}
+
+// Mapper defines the interface for mapping OTLP metric data points to Datadog format.
+// Consumer includes both sketch and raw histogram consumers - implementations can
+// use either based on configuration. Consumers can no-op methods they don't need.
+type Mapper interface {
```

Comment:
Is there a need to export this concept outside of this package? Based on the design doc, it feels like what we want is to either use the current translator or the minimal translator, right? We don't necessarily want to have custom `Mapper` implementations outside of this package. 

If that is the case, I think we can unexport this. If you want to avoid having everything on the same package we could use an `internal` package